lib/cli/src/commands/check/utils/check-matching-routes.ts

```diff
@@ -171,16 +171,15 @@ function checkQueryParameters(
 
     // Check parameter schema compatibility if schemas exist
     if (baseParam.schema && implParam.schema) {
+      // Flatten schemas before compatibility check
+      const flattenedBaseSchema = deepFlattenAllOf(baseParam.schema as OpenAPIV3.SchemaObject);
+      const flattenedImplSchema = deepFlattenAllOf(implParam.schema as OpenAPIV3.SchemaObject);
+
```

Comment:
While we were able to remove duplicate calls to `deepFlattenAllOf()` in most places, this was actually a place where we _weren't_ flattening query params but needed to -- though the likelihood of having a nested `allOf` in a query param is quite low.